peachjam_search/views.py

```diff
@@ -19,6 +24,54 @@
 CACHE_SECS = 15 * 60
 
 
+class PageQueryBackend(BaseSearchQueryBackend):
+    """Does a nested page search, and include highlights."""
+
+    @classmethod
+    def construct_search(cls, request, view, search_backend):
+        queries = []
+        for search_term in search_backend.get_search_query_params(request):
+            print(search_term)
+            queries.append(
+                Q(
+                    "nested",
+                    path="pages",
+                    query=Q(
+                        "bool",
+                        must=[
+                            SimpleQueryString(
+                                query=search_term,
+                                default_operator="and",
+                                quote_field_suffix=".exact",
+                                fields=["pages.body"],
+                            )
+                        ],
+                        should=[
+                            MatchPhrase(pages__body={"query": search_term, "slop": 2}),
+                        ],
+                    ),
+                    inner_hits={
+                        "_source": ["pages.page_num"],
+                        "highlight": {
+                            "fields": {"pages.body": {}},
+                            "pre_tags": ["<mark>"],
+                            "post_tags": ["</mark>"],
+                            "fragment_size": 80,
+                            "number_of_fragments": 2,
+                        },
+                    },
+                )
+            )
+        return queries
+
+
+class CustomSearchBackend(BaseSearchFilterBackend):
+    query_backends = [PageQueryBackend]
+
+    def get_query_backends(self, request, view):
+        return self.query_backends
```

Comment:
@ro6ley Hi I am implementing this page-based indexing but for some reason I am not getting any results with this custom backend. Not sure where the problem is. Maybe you could take a look?